### ISSUE TITLE
warn user about installing with insufficient /tmp

### ISSUE THREAD

**Issue by obigriffith** [genome/gms#14]
On the GMS-Griffith install we chose to create two partitions. One for / with 100GB and one for /data with the rest of the 3TB drive. This was on the assumption that ubuntu itself would not need much space and that large data downloads etc would be preserved on a separate partition to allow starting a fresh install easier. However, this doesn't leave much space for /tmp and prevents some jobs from launching if they request more than the available /tmp space. A simple fix for this is to just have one partition, set up a separate partition for /tmp during ubuntu install, ideally on a separate physical drive. Or, to create the appropriate symlink of /tmp to /data/tmp or the like. We should warn users about this or somehow deal with it for them.

**Comment by sakoht** [comment]
Yeah doing something special with tmp is wise.  It needs higher IO than the OS partition perhaps.

**Comment by malachig** [comment]
I don't think this symlink strategy works unfortunately.  Or at least it does not seem that simple.

/tmp seems to have special properties.  I tried creating a symlink and other things, but all of them seem to create problems for creating new builds.  Errors related to POSIX commands in workflow, inability to create allocations in /tmp, etc.  

Unsure how to proceed now...  For like the 100th time I am wishing I could control the size of my '/' mount in the virtual machine

**Comment by mcallaway** [comment]
VBoxManage modifyhd filename.vdi --resize 96000 # Vbox 4 adds resize

**Comment by malachig** [comment]
I believe this only works with 'vdi' disks.  Vagrant fires up with a 'box-disk1.vmdk', i.e. 'vmdk' format.  The resize utility does not work on these disks unfortunately...

VBoxManage: error: Resize hard disk operation for this format is not implemented yet!

It might be possible to do a convert to vmdk and then resize as described here:

http://stackoverflow.com/questions/11659005/how-to-resize-a-virtualbox-vmdk-file

What we really want is for all of this to be automated and simple for the end user so we are still trying to find a way to make that work...

**Comment by sakoht** [comment]
Instead of the symlink try mount with the -o bind.

Also, I think the Mac uses /tmp as a symlink elsewhere.  You may just have to get the permissions right.  They are something wierd for /tmp.

**Comment by malachig** [comment]
I believe this issue has been resolved in two ways:

1.) For the VM install, /tmp is now places inside a dedicated VDI with a configurable location and size.  The default setup should allow builds that need decent amounts of /tmp to succeed as long as the system has enough physical space.  The config files also act as a reminder to the user to consider this factor.

2.) In the README.md I have supplemented the System requirements section to briefly discuss /tmp requirements.
